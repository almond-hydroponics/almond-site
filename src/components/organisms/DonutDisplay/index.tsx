// highcharts
import * as Highcharts from 'highcharts/highcharts.js';
import highchartsMore from 'highcharts/highcharts-more.js';
import solidGauge from 'highcharts/modules/solid-gauge.js';
import HighchartsReact from 'highcharts-react-official';
import HighchartsExporting from 'highcharts/modules/exporting';

if (typeof Highcharts === 'object') {
	HighchartsExporting(Highcharts);
	highchartsMore(Highcharts);
	solidGauge(Highcharts);
}

interface Props {
	data: number;
}

const DonutDisplay = ({ data }: Props): JSX.Element => {
	const options = {
		chart: {
			type: 'solidgauge',
			width: 250,
			height: 250,
			style: {
				fontFamily: 'Google Sans, Roboto, Helvetica Neue, sans-serif',
			},
		},
		title: null,
		tooltip: {
			enabled: false,
		},
		pane: {
			startAngle: 0,
			endAngle: 360,
			background: [
				{
					outerRadius: '100%',
					innerRadius: '60%',
					// @ts-expect-error
					backgroundColor: Highcharts.color(Highcharts?.getOptions().colors[0])
						.setOpacity(0.3)
						.get(),
					borderWidth: 0,
				},
			],
		},
		yAxis: {
			min: 0,
			max: 100,
			lineWidth: 0,
			tickPositions: [],
		},
		plotOptions: {
			solidgauge: {
				dataLabels: {
					enabled: false,
				},
				linecap: 'round',
				stickyTracking: false,
				rounded: true,
			},
		},
		series: [
			{
				name: '',
				type: 'solidgauge',
				data: [
					{
						// @ts-expect-error
						color: Highcharts?.getOptions().colors[0],
						radius: '100%',
						innerRadius: '60%',
						y: data,
					},
				],
			},
		],
		credits: {
			enabled: false,
		},
		exporting: {
			enabled: false,
		},
	};

	return (
		<>
			<HighchartsReact highcharts={Highcharts} options={options} />
		</>
	);
};

export default DonutDisplay;
