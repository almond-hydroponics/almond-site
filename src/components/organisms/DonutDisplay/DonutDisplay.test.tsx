// react libraries
import { render } from '@testing-library/react';

// component
import DonutDisplay from './index';

describe('DonutDisplay component', () => {
	const props = {
		data: [50, 60],
		donutInfo: '',
		hoverBackgroundColor: '#1967D2',
		backgroundColor: '#1967D2',
		halfDonut: true,
	};

	it('should render correctly', () => {
		const { asFragment } = render(<DonutDisplay {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
