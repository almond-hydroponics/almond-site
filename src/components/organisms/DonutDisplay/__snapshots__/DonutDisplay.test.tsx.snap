// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutDisplay component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-6wbyuk-MuiGrid-root"
  >
    <canvas
      height="250"
      width="250"
    />
  </div>
</DocumentFragment>
`;
